control_toolbox/src/pid.cpp

```diff
@@ -383,19 +339,32 @@ double Pid::compute_command(double error, double error_dot, const double & dt_s)
   // Calculate derivative contribution to command
   d_term = gains.d_gain_ * d_error_;
 
-  // Calculate integral contribution to command
-  if (gains.antiwindup_ && gains.antiwindup_strat_ == AntiwindupStrategy::NONE)
+  if (gains.antiwindup_strat_.type == AntiWindupStrategy::UNDEFINED)
   {
-    // Prevent i_term_ from climbing higher than permitted by i_max_/i_min_
-    i_term_ = std::clamp(i_term_ + gains.i_gain_ * dt_s * p_error_, gains.i_min_, gains.i_max_);
+    throw std::runtime_error(
+      "PID: Antiwindup strategy cannot be UNDEFINED. Please set a valid antiwindup strategy.");
   }
-  else if (!gains.antiwindup_ && gains.antiwindup_strat_ == AntiwindupStrategy::NONE)
+
+  // Calculate integral contribution to command
+  const bool is_error_in_deadband_zone =
+    control_toolbox::is_zero(error, gains.antiwindup_strat_.error_deadband);
+  if (!is_error_in_deadband_zone && gains.antiwindup_strat_.type == AntiWindupStrategy::LEGACY)
```

Comment:
```suggestion
  const bool is_error_in_deadband_zone =
    control_toolbox::is_zero(error, gains.antiwindup_strat_.error_deadband);
  if (gains.antiwindup_strat_.type == AntiWindupStrategy::LEGACY)
```
should we really change the behavior here for the LEGACY?

Reply:
By default, it is `epsilon`, so mostly we are not changing the behavior unless someone explicitly sets it

Reply:
actually, I wanted to write if we want to enable to change the LEGACY behavior, as we want to remove it anyways. but I'm fine with keeping it